### PR TITLE
Added truncated context to decode exception.

### PR DESCRIPTION
**Pull request by klaussilveira**

### CODE REVIEW

**Review by damian-af — DISMISSED** [review]

**Review by melissabillias — DISMISSED** [review]

**Review by ericzucho — DISMISSED** [review]

File: src/Json.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static function encode($data): string
 
         $result = json_encode($data);
 
-        if (json_last_error() !== JSON_ERROR_NONE) {
+        if (JSON_ERROR_NONE !== json_last_error()) {
```
Comment (nesl247) [inline src/Json.php]: Why are we using yoda conditions here?

File: src/Json.php [hunk 2]
```diff
@@ -46,8 +46,15 @@ public static function decode($data)
 
         $result = json_decode($data, true);
 
-        if (json_last_error() !== JSON_ERROR_NONE) {
-            throw new LogicException(self::getLastJsonError());
+        if (JSON_ERROR_NONE !== json_last_error()) {
```
Comment (nesl247) [inline src/Json.php]: Why are we using yoda conditions?

**Review by melissabillias — DISMISSED** [review]

File: src/Json.php [hunk 3]
```diff
@@ -46,8 +46,15 @@ public static function decode($data)
...
+            $lastError = self::getLastJsonError();
+            $payload = substr($data, 0, 255);
+
+            if (strlen($data) > 255) {
+                $payload .= '... (truncated)';
+            }
+
+            throw new LogicException(sprintf('%s on \'%s\'', $lastError, $payload));
```
Comment (nesl247) [inline src/Json.php:57]: I would probably change this to `occurred on` or `occurred in`.

File: tests/JsonTest.php [hunk 4]
```diff
@@ -29,6 +31,7 @@ public function testIsHandlingInvalidDataForEncode()
 
     /**
      * @expectedException \LogicException
+     * @expectedExceptionMessage Invalid JSON: Malformed UTF-8 characters, possibly incorrectly encoded
```
Comment (nesl247) [inline tests/JsonTest.php:34]: If we are refactoring this, why not use `$this->expectException()` and such instead? It makes the test much better to read in a lot of cases and is what we use in our other applications.

**Review by nesl247 — DISMISSED** [review]

File: src/Json.php (hunk 1 above)
Comment (klaussilveira) [reply src/Json.php]: cs-fixer defaults, maybe.

File: tests/JsonTest.php (hunk 4 above)
Comment (klaussilveira) [reply tests/JsonTest.php:34]: We're not refactoring, just adding a new feature.

File: src/Json.php (hunk 3 above)
Comment (klaussilveira) [reply src/Json.php:57]: Most "Syntax Error" messages on compilers use "on", followed by the affected line and contents. Why change that?

Comment (nesl247) [reply src/Json.php:57]: Actually in this case simply changing `on` to `in` would make it follow PHP. The `occurred` part was something that popped in my head, but isn't necessary.

For example:

```
>>> json_code()
PHP Fatal error:  Call to undefined function json_code() in eval()'d code on line 1
```

File: tests/JsonTest.php (hunk 4 above)
Comment (nesl247) [reply tests/JsonTest.php:34]: You're right, it is just adding the message. However, this is a good opportunity to do a small refactoring. But I'll leave it be either way.

**Review by nesl247 — APPROVED** [review]

**Review by ericzucho — APPROVED** [review]

**Pull request merged**